omniledger/service/service.go

```diff
@@ -474,12 +480,28 @@ func (s *Service) verifySkipBlock(newID []byte, newSB *skipchain.SkipBlock) (val
 	txs := data.Transactions
 	cdb := s.getCollection(newSB.Hash)
 	for _, tx := range txs {
-		f, ok := s.verifiers[string(tx.Kind)]
-		if ok {
-			validSB = validSB && f(cdb, &tx)
+		f, exists := s.verifiers[string(tx.Kind)]
+		if !exists {
+			continue
+		}
+		if tx.Valid != f(cdb, &tx) {
+			return false
+		}
+	}
+	return true
+}
+
+// validateTransactions set the valid-flag of the transaction according to the
+// registered OmniledgerVerifiers.
+func (s *Service) validateTransactions(cdb *collectionDB, txs []Transaction) {
+	for i := range txs {
+		f, exists := s.verifiers[string(txs[i].Kind)]
+		if exists {
+			txs[i].Valid = f(cdb, &txs[i])
+		} else {
+			txs[i].Valid = true
```

Comment:
This is only done on the leader, right? 

If all the nodes have the same set of verifiers, then the leader should reject transactions that have an unknown verification function (and later on mark the node that sent the transaction as possibly invalid).

If we suppose that the set of nodes responsible for omniledger can have a diverging set of verification function, it gets more complex. At least we should set `Valid` to `false` here, as we don't know it, so we should not give our vote to make it pass. But then a node knowing that verification function might find that it should pass and reject the block.

Proposition: 
- the _leader_ is the reference for what verification functions are valid. If it receives a transaction with an invalid verification function, it simply drops it, else a node knowing it might either validate or invalidate it and reject the block.
- if a _node_ receives a transaction with an unknown verification function, it should ignore it.